### PR TITLE
refactor: enhance date fetching logic and improve game detection

### PR DESCRIPTION
**Pull request by nikosalonen**
- Updated `determine_fetch_date` to return a tuple indicating whether the date was chosen due to morning cutoff logic, improving clarity in date handling.
- Modified `handle_no_games_found` to log specific messages based on whether the date was affected by the morning cutoff, enhancing user experience during game searches.
- Adjusted the cutoff time for showing today's games to a consistent noon (12:00) across all seasons, ensuring uniform behavior year-round.
- Added unit tests to verify the new noon cutoff behavior, ensuring accurate game visibility logic.

These changes improve the accuracy and clarity of date handling in the application, enhancing the overall user experience.

## Summary by CodeRabbit

- **New Features**
  - Time-aware date resolution with deterministic time injection for testing; fetch logic now conveys pre-noon cutoff context.

- **Bug Fixes**
  - Standardized to a year-round local-noon cutoff for showing today’s games; improved early-day handling and date fallback based on cutoff context.
  - No-games handling and logging now distinguish pre-noon cutoff vs standard flow.

- **Tests**
  - Added deterministic time-aware tests for cutoff logic, date selection, and edge cases.

- **Style**
  - Minor test formatting updates.

- **Chores**
  - Project version bumped.

### CODE REVIEW

File: src/data_fetcher/api.rs [hunk 1]
```diff
@@ -174,27 +174,34 @@ fn get_team_name(team: &ScheduleTeam) -> &str {
...
+                info!(
+                    "Using yesterday's date due to pre-noon cutoff: {}",
+                    date_str
+                );
+                (date_str, true) // This was chosen due to pre-noon cutoff
+            }
         }
-    })
+    }
 }
```
[the remark this answers is not in the document]
Comment (nikosalonen) [reply src/data_fetcher/api.rs:205]: @coderabbitai push patch for 

```
src/data_fetcher/api.rs (1)
3722-3760: Strengthen tests: make determine_fetch_date assertions deterministic
```
